pyqtgraph/graphicsItems/PlotItem/PlotItem.py

```diff
@@ -284,65 +285,155 @@ def method(self, *args, **kwargs):
+
+        Return the axis instance that was removed.
+
+        """
+        entry = self.axes.pop(name, None)
+
+        if not entry:
+            return
+
+        axis = entry['item']
+        self.layout.removeItem(axis)
+        axis.scene().removeItem(axis)
+        if unlink:
+            axis.unlinkFromView()
+
+        self.update()
+
+        return axis
+
+    # NOTE: seriously.. y'all
+    # Why do we need to always have all axes created? I don't understand
+    # this at all. Everything seems to work if you just always apply the
+    # set passed to this method **EXCEPT** for some super weird reason
+    # the view box geometry still computes as though the space for the
+    # `'bottom'` axis is always there **UNLESS** you always add that
+    # axis but hide it? Why in tf would this be the case!?!?
+    def setAxisItems(
+        self,
+        # XXX: yeah yeah, i know we can't use type annots like this yet.
+        axisItems: Optional[dict[str, AxisItem]] = None,
+        add_to_layout: bool = True,
+    ):
+        """
+        Place axis items as given by `axisItems`. Initializes
+        non-existing axis items.
+
         ==============  ==========================================================================================
         **Arguments:**
         *axisItems*     Optional dictionary instructing the PlotItem to use pre-constructed items
                         for its axes. The dict keys must be axis names ('left', 'bottom', 'right', 'top')
                         and the values must be instances of AxisItem (or at least compatible with AxisItem).
         ==============  ==========================================================================================
+
         """
-        
-        if axisItems is None:
-            axisItems = {}
-        
+        axisItems = axisItems or {}
+
+        # XXX: wth is is this even saying?!?
         # Array containing visible axis items
         # Also containing potentially hidden axes, but they are not touched so it does not matter
-        visibleAxes = ['left', 'bottom']
-        visibleAxes.extend(axisItems.keys()) # Note that it does not matter that this adds
-                                             # some values to visibleAxes a second time
-        
-        for k, pos in (('top', (1,1)), ('bottom', (3,1)), ('left', (2,0)), ('right', (2,2))):
-            if k in self.axes:
-                if k not in axisItems:
-                    continue # Nothing to do here
-                
-                # Remove old axis
-                oldAxis = self.axes[k]['item']
-                self.layout.removeItem(oldAxis)
-                oldAxis.scene().removeItem(oldAxis)
-                oldAxis.unlinkFromView()
-            
+        # visibleAxes = ['left', 'bottom']
+        # Note that it does not matter that this adds
+        # some values to visibleAxes a second time
+
+        # XXX: uhhh wat^ ..?
```

Comment:
I don't understand any of these comments.
There's weird defaults for weird defaults, special cases for skipping this axis and that axis.
It's all highly confusing and doesn't represent the semantics of the method name.